pkg/operation/shoot/types.go

```diff
@@ -70,6 +71,8 @@ type Shoot struct {
 	MachineDeployments        []extensionsv1alpha1.MachineDeployment
 
 	ETCDEncryption *etcdencryption.EncryptionConfig
+
+	ResourceRefs map[string]autoscalingv1.CrossVersionObjectReference
```

Comment:
Why is this a `autoscalingv1.CrossVersionObjectReference`? 
Did I overlook something in the reviews of #2350?
```suggestion
	ResourceRefs map[string]corev1.ObjectReference
```

Reply:
I think the reason is that `autoscalingv1.CrossVersionObjectReference` only exposes `APIGroup`, `Kind`, and `Name` while `corev1.ObjectReference` exposes more fields that would remain unused.

Reply:
@tim-ebert @rfranzke Yes, `CrossVersionObjectReference` fits much better from my perspective, and also it was initially suggested in #2321. It was introduced in #2350. I don't think we would gain much my changing it to a different reference type at this point.

Reply:
Yep, right, agreed. I just wondered and wanted to double-check.